R/deparse-data-frame.R

```diff
@@ -1,20 +1,22 @@
 #' @export
-deparse.data.frame <- function(x, as_tibble = FALSE, as_tribble = FALSE, ...) {
+deparse.data.frame <-
+  function(x, as_tibble = FALSE, as_tribble = FALSE, generate_mutate = TRUE, ...) {
+
+    need_row_names <- any(row.names(x) != as.character(seq_len(nrow(x))))
```

Comment:
Also, you could use the code of `tibble::has_row_names()` to detect existence of row names.